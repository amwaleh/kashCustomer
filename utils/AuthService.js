import { loginUser } from "../utils/handler";

export default class AuthService {

  login = data => {
    return loginUser(data)
      .then(res => {
        if (res.status != 200) throw "User was not found / wrong password";
        return res.json();
      })
      .then(json => {
        this.setUserInfo(json);
      })
      .catch(e => {
        throw e;
      });
  };

  loggedIn = () => {
    const token = this.getToken();
    return !!token;
  };

  setUserInfo(idToken) {
    localStorage.setItem("user", idToken);
  }

  getToken = () => {
    return localStorage.getItem("user");
  };

  logout = () => {
    localStorage.removeItem("user");
  };
}
