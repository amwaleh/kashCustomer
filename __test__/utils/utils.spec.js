import AuthService from '../../utils/AuthService'


describe("Pages", () => {
  describe("AuthService", () => {
    let auth=''
    const mockSet = jest.spyOn(Storage.prototype, 'setItem');
    const mockGet = jest.spyOn(Storage.prototype, 'getItem');
    const mockRemoveItem = jest.spyOn(Storage.prototype, 'getItem');
    const mockJsonPromise = Promise.resolve({});
    let loginUser = jest.fn().mockResolvedValue({ status: 200, json: () => mockJsonPromise, })
    window.fetch = jest.fn().mockResolvedValue({ status: 200, json: () => mockJsonPromise, })

    beforeEach(()=>{
      loginUser.mockClear()
      window.fetch.mockClear()
      auth = new AuthService('http://local')

    })

    it("LoggedIn", ()=> {
      auth.loggedIn()
      expect(mockGet).toHaveBeenCalledTimes(1);
    });

    it("setUserInfo", ()=> {
      auth.setUserInfo('foo')
      expect(mockSet).toHaveBeenCalledTimes(1);
    });
    it("setUserInfo", ()=> {
      auth.logout()
      expect(mockRemoveItem).toHaveBeenCalledTimes(1);
    });
    it("login", done => {
       auth.login().then(()=>{
        expect(window.fetch.mock.calls).toMatchSnapshot()
        done()
      })
    })

  })



})
