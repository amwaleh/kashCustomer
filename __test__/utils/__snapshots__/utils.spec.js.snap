// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages AuthService login 1`] = `
Array [
  Array [
    "/login",
    Object {
      "body": undefined,
      "headers": Object {
        "Accept": "application/json",
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;
